hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/TokenInfoHTSSuite.java

```diff
@@ -749,6 +754,132 @@ final HapiSpec getInfoOnInvalidNonFungibleTokenFails() {
+                                .gas(1_000_000L)
+                                .hasKnownStatus(CONTRACT_REVERT_EXECUTED))))
+                .then(
+                        getTxnRecord(TOKEN_INFO_TXN).andAllChildRecords().logged(),
+                        childRecordsCheck(
+                                TOKEN_INFO_TXN,
+                                CONTRACT_REVERT_EXECUTED,
+                                recordWith().status(INVALID_TOKEN_ID)));
+    }
+
+    @HapiTest
+    final HapiSpec getInfoForNFTByFungibleTokenAddressFails() {
+        return defaultHapiSpec("getInfoForNFTByFungibleTokenAddressFails")
+                .given(
+                        cryptoCreate(TOKEN_TREASURY).balance(0L),
+                        cryptoCreate(AUTO_RENEW_ACCOUNT).balance(0L),
+                        newKeyNamed(ADMIN_KEY),
+                        newKeyNamed(SUPPLY_KEY),
+                        uploadInitCode(TOKEN_INFO_CONTRACT),
+                        contractCreate(TOKEN_INFO_CONTRACT).gas(1_000_000L),
+                        tokenCreate(PRIMARY_TOKEN_NAME)
+                                .supplyType(TokenSupplyType.FINITE)
+                                .entityMemo(MEMO)
+                                .name(PRIMARY_TOKEN_NAME)
+                                .treasury(TOKEN_TREASURY)
+                                .autoRenewAccount(AUTO_RENEW_ACCOUNT)
+                                .autoRenewPeriod(THREE_MONTHS_IN_SECONDS)
+                                .maxSupply(1000)
+                                .initialSupply(500)
+                                .adminKey(ADMIN_KEY)
+                                .supplyKey(SUPPLY_KEY)
+                                .via(CREATE_TXN))
+                .when(withOpContext((spec, opLog) -> allRunFor(
+                        spec,
+                        contractCall(
+                                        TOKEN_INFO_CONTRACT,
+                                        GET_INFORMATION_FOR_NON_FUNGIBLE_TOKEN,
+                                        HapiParserUtil.asHeadlongAddress(
+                                                asAddress(spec.registry().getTokenID(PRIMARY_TOKEN_NAME))),
+                                        1L)
+                                .via(NON_FUNGIBLE_TOKEN_INFO_TXN)
+                                .gas(1_000_000L)
+                                .hasKnownStatus(CONTRACT_REVERT_EXECUTED))))
+                .then(
+                        getTxnRecord(NON_FUNGIBLE_TOKEN_INFO_TXN)
+                                .andAllChildRecords()
+                                .logged(),
+                        childRecordsCheck(
+                                NON_FUNGIBLE_TOKEN_INFO_TXN,
+                                CONTRACT_REVERT_EXECUTED,
+                                recordWith().status(INVALID_TOKEN_NFT_SERIAL_NUMBER)));
+    }
+
+    @HapiTest
+    final HapiSpec getInfoForFungibleTokenByNFTTokenAddressFails() {
```

Comment:
This test seems reasonable to fail with INVALID_TOKEN or similar, since we use NFT. But it seems like its behaving like getTokenInfo. 
We get SUCCESS by running it against both mono and mod.

Reply:
Understood, then maybe we could change the name to "..ddressWorks" and put a comment for the behaviour, or maybe we can remove the test itself. What do you think?

Reply:
Done, updated